### DISCUSSION TITLE
Can we embed PDF files in standalone HTML files using data URIs?

### DISCUSSION THREAD

**Discussion by jmgirard** [quarto-dev/quarto-cli#10177] [Q&A]
### Description

The code below works locally but does not actually embed the "dummy.pdf" file into the HTML via data URI/URL.

[dummy.pdf](https://github.com/user-attachments/files/16041810/dummy.pdf)

```qmd
---
title: "Embed test"
format: html
embed-resources: true
resources: 
  - "dummy.pdf"
---

<object data="dummy.pdf" type="application/pdf"></object>
```

Would it be possible to enable this without serious security concerns? See [example here](https://stackoverflow.com/questions/31441454/how-to-embed-a-base64-encoded-pdf-data-uri-into-a-html-5-object-data-attribu)

**Comment by mcanouil** [comment]
First, you should not use bare html, instead use raw code block: https://quarto.org/docs/authoring/markdown-basics.html#raw-content

Use plain markdown:

```qmd
---
title: "Quarto Playground"
format: html
embed-resources: true
---

![](index.pdf)
```

See:

- https://github.com/quarto-dev/quarto-cli/discussions/5863

**Reply by jmgirard** [reply]
The support for `![](file.pdf)` and `![](dummy.pdf){width=100%}` is great. I wish that were documented better on the quarto website.

**Reply by mcanouil** [reply]
Unfortunately, Quarto cannot document everything that Pandoc supports.
See for yourself the size of Pandoc documentation: https://pandoc.org
